bin/watch-packages-and-rebuild.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+const fs = require( 'fs' );
+const path = require( 'path' );
+const { spawnSync } = require( 'child_process' );
+const chokidar = require( 'chokidar' );
+
+const packagesDirectoryPath = path.join( '.', 'packages' );
+const watcher = chokidar.watch( packagesDirectoryPath, {
+	ignored: /^\./,
+	persistent: true,
+	ignoreInitial: true,
+} );
+watcher.on( 'change', handleChange ).on( 'add', handleChange ).on( 'unlink', handleChange );
+
+let rebuildQueue = [];
+
+setTimeout( processRebuildQueue, 100 );
+
+function processRebuildQueue() {
+	rebuildQueue.map( rebuildPackage );
+	rebuildQueue = [];
+	setTimeout( processRebuildQueue, 100 );
+}
+
+function handleChange( filePath ) {
+	if ( isPathIgnored( filePath ) ) {
+		return;
+	}
+	console.log( 'heard change in', filePath );
+	const packageName = getPackageNameFromFilePath( filePath );
+	if ( packageName && ! rebuildQueue.includes( packageName ) ) {
+		rebuildQueue.push( packageName );
+	}
+}
+
+function isPathIgnored( filePath ) {
+	const filePathPieces = filePath.split( path.sep );
+	if ( filePathPieces.length < 2 ) {
+		return true;
+	}
+	const ignoredPaths = [ 'dist', 'test', 'tests' ];
+	if ( ignoredPaths.find( ( ignored ) => filePathPieces.includes( ignored ) ) ) {
+		return true;
+	}
+	const ignoredExtensions = [ '.css', '.md', '.d.ts' ];
+	if ( ignoredExtensions.find( ( ignored ) => filePath.endsWith( ignored ) ) ) {
+		return true;
+	}
+	return false;
+}
```

Comment:
`chokidar` does all this for you [out of the box](https://github.com/paulmillr/chokidar#path-filtering)!

Reply:
I wasn't sure how to use the `ignored` property to match a directory in the path like `dist` without including the slash (`/`) character (which would break cross-platform). Is there some way to do it?

Reply:
`ignored` says it follows [anymatch](https://github.com/micromatch/anymatch) which reads:

> Note: This module has Bash-parity, please be aware that Windows-style backslashes are not supported as separators.

That's why I made my own function.

Reply:
Great research. The only thing I would do is move `ignoredPaths` and `ignoredExtensions` to the top of the file so people can maintain them later without understanding the whole file.